### PR TITLE
Solvent representation

### PR DESCRIPTION
**Pull request by yunsiechung**
I added a new solvent attribute called molecule that stores the solvent's molecular structure and a new Species attribute called isSolvent which gives Boolean True if the species is the solvent and False if not. 

In main.py, I added new codes that check,  for every liquid phase reactions, whether any of the initial species is the solvent. If none of the initial species is the solvent, then it raises Exception with the error message. 

If the RMGPy-database is not updated (with the one containing my latest pull request), RMG will determine whether the species is the solvent or not based on the solvent's string name

If the RMGPy-database is updated with the version containing my latest pull request, RMG will determine whether the species is the solvent or not based on the solvent's molecular structures.

### CODE REVIEW

**Comment by nickvandewiele** [comment]
I'd advise you to add multiple unit tests for this.

File: rmgpy/data/solvation.py [hunk 1]
```diff
@@ -271,22 +271,25 @@ def __init__(self, label='', name='', shortDesc='', longDesc=''):
...
+        mol = molecule
+        if molecule is not None:
             try:
-                mol = Species().fromAdjacencyList(molecule)
+                mol = Species().fromSMILES(molecule)
             except:
-                logging.error("Can't understand '{0}' in solute library '{1}'".format(molecule,self.name))
-                raise
-        mol.generateResonanceIsomers()
+                logging.debug("Solvent '{0}' does have a valid SMILES '{1}'" .format(label, molecule))
```
Comment (bslakman) [inline rmgpy/data/solvation.py]: Should this say "does not have a valid SMILES"?

Comment (yunsiechung) [inline rmgpy/data/solvation.py]: yeah you are right. I will correct that.

**Comment by yunsiechung** [comment]
I will make some unit test and update this branch!

**Comment by yunsiechung** [comment]
I added two unittests in the rmgpy/data/solvationTest.py to test
1) it can determine whether the solvent is listed as the initial species or not in various cases
2) it can correctly load the solvent molecular structure when the solvent database contains the molecule attribute, and knows leave the solvent molecular structure as None when the solvent molecular structures are not provided in the database.

**Comment by yunsiechung** [comment]
@bslakman I am finally done with modifying this branch. Do you mind checking my codes and if they look fine, merge it?

I tried my best to describe everything in details in rmgpy/data/solvationTest.py. If you just look at the last two tests in that file, it should be clear.

Let me know if there's anything wrong. Thank you!

**Comment by rwest** [comment]
May I suggest an interactive rebase to squash the typo-fixing commits?

**Comment by yunsiechung** [comment]
@rwest Thanks for the suggestion. I just did that!

**Comment by rwest** [comment]
Quick glance looks good to me, but do any documentation or example input files need updating?
eg. a note near the top of `documentation/source/users/rmg/liquids.rst` that the solvent needs to be a reactant?

**Comment by yunsiechung** [comment]
Example input files do not need updating, but I will update the documentation.

**Comment by yunsiechung** [comment]
@bslakman  Could you merge this? Thank you!

**Comment by yunsiechung** [comment]
@bslakman Could you do the final check and merge this? Thank you!

**Pull request merged**